mlflow/cli/eval_utils.py

```diff
@@ -0,0 +1,188 @@
+"""
+Utility functions for trace evaluation output formatting.
+"""
+
+from typing import Any
+
+import click
+import pandas as pd
+
+from mlflow.exceptions import MlflowException
+from mlflow.genai.scorers import get_all_scorers, get_scorer
+
+
+def resolve_scorers(scorer_names: list[str], experiment_id: str) -> list[Any]:
+    """
+    Resolve scorer names to scorer objects.
+
+    Checks built-in scorers first, then registered scorers.
+
+    Args:
+        scorer_names: List of scorer names to resolve
+        experiment_id: Experiment ID for looking up registered scorers
+
+    Returns:
+        List of resolved scorer objects
+
+    Raises:
+        click.UsageError: If a scorer is not found or no valid scorers specified
+    """
+    resolved_scorers = []
+    builtin_scorers = get_all_scorers()
+    builtin_scorer_map = {scorer.__class__.__name__: scorer for scorer in builtin_scorers}
+
+    for scorer_name in scorer_names:
+        # Check if it's a built-in scorer
+        if scorer_name in builtin_scorer_map:
+            resolved_scorers.append(builtin_scorer_map[scorer_name])
+        else:
+            # Try to get it as a registered scorer
+            try:
+                registered_scorer = get_scorer(name=scorer_name, experiment_id=experiment_id)
+                resolved_scorers.append(registered_scorer)
+            except MlflowException:
+                available_builtin = ", ".join(builtin_scorer_map.keys())
+                raise click.UsageError(
+                    f"Scorer '{scorer_name}' not found. "
+                    f"Available built-in scorers: {available_builtin}. "
+                    f"Or register a custom scorer first."
+                )
+
+    if not resolved_scorers:
+        raise click.UsageError("No valid scorers specified")
+
+    return resolved_scorers
+
+
+def extract_assessments_from_results(
+    results_df: pd.DataFrame, evaluation_run_id: str
+) -> list[dict[str, Any]]:
+    """
+    Extract assessments from evaluation results DataFrame.
+
+    The evaluate() function returns results with a DataFrame that contains
+    an 'assessments' column. Each row has a list of assessment dictionaries
+    with metadata including 'mlflow.assessment.sourceRunId' that we use to
+    filter assessments from this specific evaluation run.
+
+    Args:
+        results_df: DataFrame from evaluate() results containing assessments column
+        evaluation_run_id: The MLflow run ID from the evaluation that generated the assessments
+
+    Returns:
+        List of dictionaries with trace_id and assessments
+    """
+    output_data = []
+
+    for _, row in results_df.iterrows():
+        trace_id = row["trace_id"]
+        trace_result = {"trace_id": trace_id, "assessments": []}
```

Comment:
Nit - is there a more descriptive name we can use for this variable? `trace_assessments`?